lib/perl/Genome/Model/Tools/Analysis/Concordance.pm

```diff
@@ -209,28 +215,53 @@ sub execute {
 
     my (@norm_total, @norm_snp, @pre_total, @pre_snp);
 
+
+    #now read the genotypes in, calculate matches
+    my %genotypes;
+
     open (my $IN1,'<',$norm) or die "$!"; #open 1st file
     while (<$IN1>) {
-        chomp;                            #remove newline from each row of the 1st file
-        my @field = split(/\t/);          #read line from 1st file
+        chomp;
+        my @field = split(/\t/);
         if ($field[13] ne "NA"){
             push(@norm_total,join('_',$field[0],$field[1]));
             push(@norm_snp,join('_',$field[0],$field[1],$field[13]));
         }
-    }  
+        if($self->output_genotypes){
+            $genotypes{join("\t",(@field[0..2]))}{"samp1"} = $field[13];
+        }
+    }
     close $IN1;
 
     open (my $IN2,'<',$pre) or die "$!";  #open 2nd file
     while (<$IN2>) {
-        chomp;                            #remove newline from each row of the 2nd file
-        my @field = split(/\t/);          #read line from 2nd file
+        chomp;
+        my @field = split(/\t/);
         if ($field[13] ne "NA"){
             push(@pre_total,join('_',$field[0],$field[1]));
             push(@pre_snp,join('_',$field[0],$field[1],$field[13]));
         }
-    }  
+        if($self->output_genotypes){
+            $genotypes{join("\t",(@field[0..2]))}{"samp2"} = $field[13];
+        }
+    }
     close $IN2;
 
+
+    #output a file with all genotypes, if specified
+    if($self->output_genotypes){
+        my $outfile = Genome::Sys->open_file_for_writing($self->output_genotypes);
+        $outfile->print(join("\t",("Chr","Pos","ReferenceBase","Sample1","Sample2")) . "\n");
+        for my $pos (keys(%genotypes)){
```

Comment:
The order from `keys` is not guaranteed to be consistent.  You may want to sort these before writing them out.